debezium-server/debezium-server-pubsub/src/test/java/io/debezium/server/pubsub/PubSubIT.java

```diff
@@ -89,18 +112,76 @@ public void receiveMessage(PubsubMessage message, AckReplyConsumer consumer) {
 
     void setupDependencies(@Observes ConnectorStartedEvent event) throws IOException {
         Testing.Print.enable();
-
-        try (SubscriptionAdminClient subscriptionAdminClient = SubscriptionAdminClient.create()) {
-            final TopicName topic = TopicName.ofProjectTopicName(ServiceOptions.getDefaultProjectId(), STREAM_NAME);
+        
+        trySetupEmulatorChannel();
+        
+        try (TopicAdminClient topicAdminClient = createTopicAdminClient()) {
+            Topic topic = topicAdminClient.createTopic(topicName);
+            Testing.print("Created topic: " + topic.getName());
+        }
+        catch (AlreadyExistsException e) {
```

Comment:
Under which condition this and the next call will throw the exception?

Reply:
If the topic or sub already exists. If something fails during the teardown of the test, then those resources will not be cleaned up and tests would then fail. So this makes the test a bit more robust. 
Btw, I haven't figured out how to check if the resource is already there. I would have preferred somethings like 
```
if resource doe not exists
   create it
```
rather than having to rely on exceptions.

Reply:
@calinilie ok, this makes sense. And for the test code it is fine. Just one question, will not be the test results affected by the existing content of the topic? If yes then it would probably be better to follow - delete/create pattern when delete could fail but create must pass.